interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -83,6 +95,61 @@ func (p *Prompt) MarshalJSON() ([]byte, error) {
 	return json.Marshal(toMarshal)
 }
 
+func (p *Prompt) sendReply(outcome prompting.OutcomeType) error {
+	// Reply with any permissions which were previously allowed
+	// If outcome is allow, then reply by allowing all originally-requested
+	// permissions. If outcome is deny, only allow permissions which were
+	// originally requested but have since been allowed by rules, and deny any
+	// remaining permissions.
+	allowedPermission := responseForInterfaceConstraintsOutcome(p.Interface, p.Constraints, outcome)
+	return p.sendReplyWithPermission(allowedPermission)
+}
+
+func responseForInterfaceConstraintsOutcome(iface string, constraints *promptConstraints, outcome prompting.OutcomeType) any {
+	allow, err := outcome.AsBool()
+	if err != nil {
+		// This should not occur, but if so, default to deny
+		allow = false
+		logger.Debugf("%v", err)
```

Comment:
Nitpick, can you make this `Noticef` so that it shows up as an error in logs?
```suggestion
		logger.Debugf("failed to compute prompting outcome: %v", err)
```